### PR TITLE
Add unit tests for remaining nidigital HRAM error cases

### PR DESCRIPTION
**Pull request by ni-jfitzger**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [x] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Adds unit tests for several simple nidigital HRAM fetch cases that cannot currently be covered by system tests with simulated hardware.  This change adds the following unit tests:

* `test_fetch_history_ram_cycle_information_samples_to_read_too_much`

### List issues fixed by this Pull Request below, if any.

* #1273 is partially fixed by this PR.

### What testing has been done?

New unit tests have been executed with nidigital 19.0.1.

### CODE REVIEW

**Comment by sbethur** [comment]
ok to test

File: src/nidigital/unit_tests/test_nidigital.py [hunk 1]
```diff
@@ -36,11 +36,68 @@ def teardown_method(self, method):
...
 
         self.patched_library.niDigital_GetHistoryRAMSampleCount.side_effect = self.side_effects_helper.niDigital_GetHistoryRAMSampleCount
         self.side_effects_helper['GetHistoryRAMSampleCount']['sampleCount'] = 7
         with nidigital.Session('') as session:
             with pytest.raises(ValueError, match='position: Specified value = 8, Maximum value = 6.'):
                 session.sites[1].fetch_history_ram_cycle_information(position=8, samples_to_read=-1)
+
+    # TODO(sbethur): When nidigital driver provides better simulation support (internal bug# 992370),
+    #  this test should be converted to a system test.  
+    def test_fetch_history_ram_cycle_information_is_finite_invalid(self):
```
Comment (sbethur) [inline src/nidigital/unit_tests/test_nidigital.py]: This test can be run on simulated session as system test. You can remove this unit test.

File: src/nidigital/unit_tests/test_nidigital.py [hunk 2]
```diff
@@ -36,11 +36,68 @@ def teardown_method(self, method):
...
+            assert session.sites[1].get_history_ram_sample_count() == 7
+
+            expected_error_description = (
+                'position: Specified value = 3, samples_to_read: Specified value = 5; Samples available = 4.')
+            with pytest.raises(ValueError, match=expected_error_description):
+                session.sites[1].fetch_history_ram_cycle_information(position=3, samples_to_read=5)
+
+    # TODO(sbethur): When nidigital driver provides better simulation support (internal bug# 992370),
+    #  this test should be converted to a system test.  
+    def test_fetch_history_ram_cycle_information_samples_to_read_zero(self):
```
Comment (sbethur) [inline src/nidigital/unit_tests/test_nidigital.py]: This test can be run on simulated session as system test. You can remove this unit test.

File: generated/nidigital/nidigital/unit_tests/test_nidigital.py [hunk 3]
```diff
@@ -36,11 +36,68 @@ def teardown_method(self, method):
...
+                session.sites[1].fetch_history_ram_cycle_information(position=0, samples_to_read=-1)
+
+    # TODO(sbethur): When nidigital driver provides better simulation support (internal bug# 992370),
+    #  this test should be converted to a system test.  
+    def test_fetch_history_ram_cycle_information_samples_to_read_too_much(self):
+
+        self.patched_library.niDigital_GetHistoryRAMSampleCount.side_effect = self.side_effects_helper.niDigital_GetHistoryRAMSampleCount
+        self.side_effects_helper['GetHistoryRAMSampleCount']['sampleCount'] = 7
+        self.patched_library.niDigital_GetAttributeViBoolean.side_effect = self.side_effects_helper.niDigital_GetAttributeViBoolean
+        self.side_effects_helper['GetAttributeViBoolean']['value'] = True # history_ram_number_of_samples_is_finite
```
Comment (sbethur) [inline generated/nidigital/nidigital/unit_tests/test_nidigital.py]: linter failed on my machine:
```
flake8 run-test: commands[3] | flake8 --config=./tox.ini generated/
generated/nidigital/nidigital/unit_tests/test_nidigital.py:65:75: E261 at least two spaces before inline comment
        self.side_effects_helper['GetAttributeViBoolean']['value'] = False # history_ram_number_of_samples_is_finite
                                                                          ^
generated/nidigital/nidigital/unit_tests/test_nidigital.py:81:74: E261 at least two spaces before inline comment
        self.side_effects_helper['GetAttributeViBoolean']['value'] = True # history_ram_number_of_samples_is_finite
                                                                         ^
generated/nidigital/nidigital/unit_tests/test_nidigital.py:103:48: W292 no newline at end of file
        assert len(history_ram_cycle_info) == 0                                               ^
```

Fix it and ensure linter doesn't flag anything else.

File: src/nidigital/unit_tests/test_nidigital.py [hunk 4]
```diff
@@ -36,11 +36,68 @@ def teardown_method(self, method):
     # API Tests
 
     # TODO(sbethur): When nidigital driver provides better simulation support (internal bug# 992370),
-    #  this test should be converted to a system test (GitHub issue# 1353).
+    #  this test should be converted to a system test.
```
[the remark this answers is not in the document]
Comment (ni-jfitzger) [reply src/nidigital/unit_tests/test_nidigital.py]: Issue # Added to all remaining tests

File: src/nidigital/unit_tests/test_nidigital.py (hunk 1 above)
Comment (ni-jfitzger) [reply src/nidigital/unit_tests/test_nidigital.py]: Removed

File: src/nidigital/unit_tests/test_nidigital.py [hunk 5]
```diff
@@ -36,11 +36,68 @@ def teardown_method(self, method):
...
+    def test_fetch_history_ram_cycle_information_position_negative(self):
```
[the remark this answers is not in the document]
Comment (ni-jfitzger) [reply src/nidigital/unit_tests/test_nidigital.py]: Removed

File: generated/nidigital/nidigital/unit_tests/test_nidigital.py (hunk 3 above)
Comment (ni-jfitzger) [reply generated/nidigital/nidigital/unit_tests/test_nidigital.py]: Fixed.

File: src/nidigital/unit_tests/test_nidigital.py (hunk 2 above)
Comment (ni-jfitzger) [reply src/nidigital/unit_tests/test_nidigital.py]: Removed

**Pull request merged**